pkg/idp/provider/provider.go

```diff
@@ -29,6 +30,11 @@ type ProviderCredential struct {
 	Credentials map[string]string `yaml:"credentials"`
 }
 
+type ProviderApp struct {
```

Comment:
Instead of just the connector, we return this struct now that also contains the time of secret expiry. We can also use it to add more properties later in case that's needed.